server/routes/activities/allocation-dashboard/handlers/allocationDashboard.ts

```diff
@@ -59,6 +62,31 @@ export default class AllocationDashboardRoutes {
       this.getCandidates(+activityId, filters, +req.query.page, user),
     ])
 
+    const daysOfWeek = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+
+    const slots: { [weekNumber: string]: Slots } = {}
+    activity.schedules[0].slots.forEach(slot => {
+      daysOfWeek.forEach(day => {
+        const dayLowerCase = day.toLowerCase()
+        slots[slot.weekNumber] ??= {
+          days: [],
+        }
+        slots[slot.weekNumber][`timeSlots${day}`] ??= []
+
+        if (slot[`${dayLowerCase}Flag`]) {
+          slots[slot.weekNumber].days.push(dayLowerCase)
+          slots[slot.weekNumber][`timeSlots${day}`].push(getTimeSlotFromTime(slot.startTime).toUpperCase())
+        }
+      })
+    })
```

Comment:
It's probably fine as is, but as a note it might be worth making this into a util function since this logic is also in `activity.ts`